engine/apps/schedules/ical_utils.py

```diff
@@ -377,11 +377,11 @@ def get_oncall_users_for_multiple_schedules(
         return {}
 
     # Get on-call users
-    oncall_users = {}
+    oncall_users: SchedulesOnCallUsers = {}
     for schedule in schedules:
         # pass user list to list_users_to_notify_from_ical
         schedule_oncall_users = list_users_to_notify_from_ical(schedule, events_datetime=events_datetime)
-        oncall_users.update({schedule.pk: schedule_oncall_users})
+        oncall_users.update({schedule: schedule_oncall_users})
```

Comment:
it's much easier here to just return the schedule object in case we need to reference other fields on it (like in the `TeamSerializer` where we need to reference `schedule.team`)